articles/discord-bot-node/index.md

```diff
@@ -0,0 +1,143 @@
+
+Make a file called `index.js` and open it up in your text editor of choice. There's some basic boilerplate code that you'll need to get started.
+
+```javascript
+const Discord = require("discord.js"); // imports the discord library
+const fs = require("fs"); // imports the file io library
+
+const client = new Discord.Client(); // creates a discord client
+const token = fs.readFileSync("token.txt").toString(); // gets your token from the file
+
+client.once("ready", () => { // prints "Ready!" to the console once the bot is online
+	console.log("Ready!");
+});
+
+client.login(token); // starts the bot up
+```
+
+You can run this by using `node index.js`. Once you’ve done that it should look something like this:
+
+![The word “ready” appears in the command prompt](/engineering-education/discord-bot-node/discordbot9.png)
+
+## Creating Commands
+
+That's cool, but we want the bot to actually do something. The most common example is [ping-pong](https://www.youtube.com/watch?v=DEqrCI1018I), but we should do something a little more interesting. Let's generate random numbers. Whenever a user types in "?random", let's make the bot reply with a random number. We just need to add some more code above the `client.login()` call.
+
+```javascript
+client.on("message", message => { // runs whenever a message is sent
+    if (message.content === "?random") { // checks if the message says "?random"
+        const number = Math.random(); // generates a random number
+        message.channel.send(number.toString()); // sends a message to the channel with the number
+    }
+});
+```
+
+We did it! Albeit, in a [less than optimal way](http://www0.cs.ucl.ac.uk/staff/d.jones/GoodPracticeRNG.pdf).
+
+We can add more commands by just adding more `if` statements. This can get messy though. Let's come up with a better command structure. We can use a Map to store strings as keys and functions as values.
+
+```javascript
+function random(message) {
+    const number = Math.random(); // generates a random number
+    message.channel.send(number.toString()); // sends a message to the channel with the number
+}
+
+let commands = new Map();
+commands.set("random", random);
+
+client.on("message", message => {
+    if (message.content[0] === '?') {
+        const command = message.content.split(" ")[0].substr(1); // gets the command name
+        if (commands.has(command)) { // checks if the map contains the command
+            commands.get(command)(message) // runs the command
+        }
+    }
+});
+```
+
+![I say "?random", and the bot replies with a random number](/engineering-education/discord-bot-node/discordbot0.png)
+
+There. Now to add more commands, you just add them to `commands`.
+
+## Bot Ideas
+Now that you know how to make a Discord bot, it would nice if you would actually make one. If you need a list of Discord Bot ideas, I've come up with a few already for [my blog](https://botahamec.github.io/posts/20_06_03_project_ideas/). Here are a few more though:
```

Comment:
I would recap this article here.

Also, the first sentence here has a typo and sounds a bit judgy...fyi.

Reply:
Looking back at it, yeah it is judgy. Call me crazy, but where is the typo?

Reply:
@botahamec - I think it was a small missing word:
"it would nice if you would actually make one."
"it would be nice if you would..."

🚀